### ISSUE TITLE
Implement TranslatorBagInterface in Translator class

### ISSUE THREAD

**Issue by koertho** [contao/core-bundle#1674]
The decorated translator service, introduced in Contao 4.5, doesn't implement the TranslatorBagInterface which the original Symfony service implements. Cause some extension rely on this service and the implementation (so we are, and we are working with Contao 4.4, but got some errors from 4.5 users), this should be fixed (also for 4.5).

See [Translator class](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/Translator.php) and [TranslatorBagInterface](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/TranslatorBagInterface.php)

**Comment by dmolineus** [comment]
The service uses only the TranslatorInterface. You shouldn't rely on the actual implementation but on the interface.

https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml#L23

If you need to access the catalogue you maybe use the `translator.default` service.

**Comment by leofeyer** [comment]
@ausi What do you think?

**Comment by ausi** [comment]
Because we decorate the original Symfony service, I think we should probably add the bag interface and forward the method call.  
Should be as simple as adding:

```diff
+ use Symfony\Component\Translation\TranslatorBagInterface;
  
- class Translator implements TranslatorInterface
+ class Translator implements TranslatorInterface, TranslatorBagInterface
```

and:

```diff
+ public function getCatalogue($locale = null)
+ {
+     return $this->translator->getCatalogue($locale);
+ }
```

**Comment by ausi** [comment]
The correct way to use it in a third party bundle could be something like:

```php
if ($transaltor instanceof TranslatorBagInterface) { … }
```

**Comment by dmolineus** [comment]
Yes, and Contao could provide two decorator one implementing the `TranslatorBagInterface` and one without depending on the actual service. 

It's basically what I do in my [backport](https://github.com/netzmacht/contao-toolkit/blob/3f731a68e75a19b87fc0b5f631ed1cc74ddf2605/src/Bundle/DependencyInjection/Compiler/TranslatorPass.php#L45-L59) for Contao 4.4: 

```php
        $definition      = $container->findDefinition('translator');
        $translatorClass = $container->getParameterBag()->resolveValue($definition->getClass());
        $decoratorClass  = is_subclass_of($translatorClass, TranslatorBag::class)
            ? LangArrayTranslatorBagTranslator::class
            : LangArrayTranslator::class;

        $definition = new Definition(
            $decoratorClass,
            [
                new Reference('netzmacht.contao_toolkit.translation.translator.inner'),
                new Reference('contao.framework')
            ]
        );

        $definition->setDecoratedService('translator');
```

**Comment by aschempp** [comment]
We should totally implement that interface like @ausi mentioned in 
https://github.com/contao/core-bundle/issues/1674#issuecomment-414838525. Everything else seems wrong to me

**Comment by leofeyer** [comment]
Implemented in 28404321f37587231fb1da978d55d741f7b8a158.
